src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int GROW_INDEX = 2;
+    private Node<K,V>[] table;
+    private int size;
+    private int threashold;
+    private Node<K,V> newNode;
+    private Node<K,V> boofNode;
+    private int index;
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threashold) {
+            resize();
+        }
+        newNode = new Node<>(hash(key),key,value,null);
+        index = Math.abs(newNode.hash % table.length);
+        if (table[index] != null) {
+            for (Node<K,V> nodeCount = table[index];nodeCount != null;nodeCount = nodeCount.next) {
+                if (Objects.equals(nodeCount.key, newNode.key)) {
+                    nodeCount.value = newNode.value;
+                    break;
+                } else if (nodeCount.next == null) {
+                    nodeCount.next = newNode;
+                    size++;
+                    break;
+                }
+            }
+        } else {
+            table[index] = newNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+        for (int i = 0;i < table.length; i++) {
+            if (table[i] != null) {
+                if (Objects.equals(table[i].key,key)) {
+                    return table[i].value;
+                }
+                boofNode = table[i].next;
+                while (boofNode != null) {
+                    if (Objects.equals(boofNode.key,key)) {
+                        return boofNode.value;
+                    }
+                    boofNode = boofNode.next;
+                }
+            }
+        }
```

Comment:
we don't need loop inside the loop here, let's simplify this method and instead of first loop let's find needed index using hash and then use only one while loop